web/blog/2023-10-17-from-candidate-to-colleague.md

```diff
@@ -0,0 +1,604 @@
+    - Codebase reusability: The TradeArea component was successfully made reusable, demonstrating the potential for other complex components to be transformed into reusable ones.
+    - Good code quality and structure: Functions and variables fulfill their intended purposes and have appropriate names (to check the importance of naming, you can give a read [here](https://dev.to/wasp/why-naming-is-1-skill-for-writing-clean-code-4a5p)).
+      - Here are some examples of that:
+        - `loadPokémons()`, `registerTrade()`, `setSelectedPokémon()` are all good naming examples in which we can get the basic sense of what the function does just by reading their name.
+    - Responsiveness: The UI is designed with a mobile-first approach (thanks to [TailwindCSS](https://tailwindcss.com/)).
+  - Bad:
+
+    - Lack of unit tests for complex behaviors.
+    - Consider using a more user-friendly method to display all Pokémon options, rather than a simple select with 200 options.
+      - For example: There are alternative [select components](https://react-select.com/home) available that could be used.
+      - You can see that this select has an input attached to it (which would bring a much better UX for the user if he wanted a Pokémon who is on the end of the list)
+
+      - ![Select Example](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/2kb1pmn2hpmvulb893wz.png)
+    - The front-end currently parses the JSON.stringify data from the back-end. It would be better if the back-end sent the data already correctly parsed.
+
+- Back-end and database:
+  - Good:
+    - Clear separation of scopes: The structures are well-organized, and functions are appropriately placed.
+    - Error handling: A few error handlers are in place, enhancing the application's resiliency.
+      - We have some private insertions validations in place (e.g. `if (!user){ throw new HttpError(401); }`)
+      - And also some basic auth validations that the auth method provides by itself via the Wasp Framework.
+    - Effective use of async/await and promise handling.
+  - Bad:
+    - Suboptimal choice of database: The absence of JSON in our database is a drawback, but it's commendable that a creative solution was found for this issue.
+    - Lack of unit tests.
+
+Overall, we have a simple application that functions well. There is evident consideration given to the user interface and user experience, which is a positive aspect. The candidate also demonstrated creativity in resolving encountered issues. However, it would have been beneficial for the candidate to include unit tests and consider a more suitable database choice. Nevertheless, good work overall! 👍
+
+## Conclusion
+
+Just in case you haven't starred [Wasp's repository](https://www.github.com/wasp-lang/wasp), I would recommend doing so! It's a great repository for you to test it out on a full-stack application!
+
+{% cta https://www.github.com/wasp-lang/wasp %} ⭐️ Give it a star! ⭐️ {% endcta %}
```

Comment:
only works for dev.to. this needs to be turned into a link

Reply:
Nice catch! Fixed 👍